src/ops.rs

```diff
@@ -197,16 +217,47 @@ pub fn squash(
     Ok(new_id)
 }
 
-// From git2 crate
-#[cfg(unix)]
-fn bytes2path(b: &[u8]) -> &std::path::Path {
-    use std::os::unix::prelude::*;
-    std::path::Path::new(std::ffi::OsStr::from_bytes(b))
-}
+/// Filter the directory tree of `commit_id` to just the specified file paths
+///
+/// Performance of some git2 operations scale with the number of files within the repo.  Operating on just the
+/// subset of files should dramatically speed up these operations
+pub fn filter_commit(
+    repo: &git2::Repository,
+    commit_id: git2::Oid,
+    parent_id: Option<git2::Oid>,
+    filtered_paths: &[&std::path::Path],
+) -> Result<git2::Oid, git2::Error> {
+    let commit = repo.find_commit(commit_id)?;
+    let tree = commit.tree()?;
+    let dehydrated_tree_oid = crate::tree::filter_tree(repo, &tree, filtered_paths)?;
+    let dehydrated_tree = repo.find_tree(dehydrated_tree_oid)?;
+
+    let signature = git2::Signature::new(
+        "git-branchless",
+        "git-branchless@example.com",
+        &git2::Time::new(0, 0),
+    )?;
+    let message = format!(
+        "generated by git-branchless: temporary dehydrated commit \
+                \
+                This commit was originally: {}",
```

Comment:
Probably worth changing `git-branchless` to `git2-ext`?

Reply:
As its not adding much atm I've just removed filter_commit